extensions/aerosol_optics/mo_aerosol_optics.F90

```diff
@@ -0,0 +1,681 @@
+             this%aero_ocar_rh_ssa(nrh, nband), &
+             this%aero_ocar_rh_asy(nrh, nband))
+
+    ! Load LUT coefficients
+    !$acc kernels
+    !$omp target
+    this%aero_dust_ext = aero_dust_ext
+    this%aero_dust_ssa = aero_dust_ssa
+    this%aero_dust_asy = aero_dust_asy
+    this%aero_salt_ext = aero_salt_ext
+    this%aero_salt_ssa = aero_salt_ssa
+    this%aero_salt_asy = aero_salt_asy
+    this%aero_sulf_ext = aero_sulf_ext
+    this%aero_sulf_ssa = aero_sulf_ssa
+    this%aero_sulf_asy = aero_sulf_asy
+    this%aero_bcar_ext = aero_bcar_ext
+    this%aero_bcar_ssa = aero_bcar_ssa
+    this%aero_bcar_asy = aero_bcar_asy
+    this%aero_bcar_rh_ext = aero_bcar_rh_ext
+    this%aero_bcar_rh_ssa = aero_bcar_rh_ssa
+    this%aero_bcar_rh_asy = aero_bcar_rh_asy
+    this%aero_ocar_ext = aero_ocar_ext
+    this%aero_ocar_ssa = aero_ocar_ssa
+    this%aero_ocar_asy = aero_ocar_asy
+    this%aero_ocar_rh_ext = aero_ocar_rh_ext
+    this%aero_ocar_rh_ssa = aero_ocar_rh_ssa
+    this%aero_ocar_rh_asy = aero_ocar_rh_asy
+    !$acc end kernels
+    !$omp end target
+    !
+  end function load_lut
+  !--------------------------------------------------------------------------------------------------------------------
+  !
+  ! Finalize
+  !
+  !--------------------------------------------------------------------------------------------------------------------
+  subroutine finalize(this)
+    class(ty_aerosol_optics), intent(inout) :: this
+
+    ! Lookup table aerosol optics interpolation arrays
+    if(allocated(this%aero_bin_lo)) then
+
+      deallocate(this%aero_bin_lo, this%aero_bin_hi, this%aero_bin_eff, &
+                 this%aero_rh)
+    end if
+
+    ! Lookup table aerosol optics coefficients
+    if(allocated(this%aero_dust_ext)) then
+
+      deallocate(this%aero_dust_ext, this%aero_dust_ssa, this%aero_dust_asy, &
+                 this%aero_salt_ext, this%aero_salt_ssa, this%aero_salt_asy, &
+                 this%aero_sulf_ext, this%aero_sulf_ssa, this%aero_sulf_asy, &
+                 this%aero_bcar_ext, this%aero_bcar_ssa, this%aero_bcar_asy, &
+                 this%aero_bcar_rh_ext, this%aero_bcar_rh_ssa, this%aero_bcar_rh_asy, &
+                 this%aero_ocar_ext, this%aero_ocar_ssa, this%aero_ocar_asy, &
+                 this%aero_ocar_rh_ext, this%aero_ocar_rh_ssa, this%aero_ocar_rh_asy)
+    end if
+
+  end subroutine finalize
+  ! ------------------------------------------------------------------------------
+  !
+  ! Derive aerosol optical properties from provided aerosol input properties
+  !
+  ! ------------------------------------------------------------------------------
+  !
+  ! Compute single-scattering properties
+  !
+  function aerosol_optics(this, aero_type, aero_size, aero_mass, relhum, &
+                          optical_props) result(error_msg)
+
+    class(ty_aerosol_optics), &
+              intent(in  ) :: this
+    integer,  intent(in  ) :: aero_type(:,:)   ! MERRA2/GOCART aerosol type
```

Comment:
@mjiacono Please add comments to clarify the physical meaning of the dimensions here (I'm guessing col, level). Is only one aerosol type permitted per location?

Reply:
Added array dimensions for aero_type, aero_size, aero_mass, and relhum. All are (ncol,nlay). One aerosol type is allowed per layer, but different types can be selected across layers.